Common/Util/RateLimit/BusyWaitSleepStrategy.cs

```diff
@@ -0,0 +1,36 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System.Threading;
+
+namespace QuantConnect.Util.RateLimit
+{
+    /// <summary>
+    /// Provides a CPU intensive means of waiting for more tokens to be available in <see cref="ITokenBucket"/>.
+    /// This strategy is only viable when the requested number of tokens is expected to become available in an
+    /// extremely short period of time. This implementation aims to keep the current thread executing to prevent
+    /// potential content switches arising from a thread yielding or sleeping strategy.
+    /// </summary>
+    public class BusyWaitSleepStrategy : ISleepStrategy
```

Comment:
If this class is only used in tests it might make sense to move it there

Reply:
It's actually part of the ported code from the referenced github repository. That being said, I think it rounds out the `ISleepStrategy` space and to be honest, I _really_ like the idea of an `ISleepStrategy`. Think about all the hard coded `Thread.Sleep( _n_ )` everywhere. I'm happy to move it to tests if there are strong opinions here, but in my opinion, it doesn't negatively impact anything and  rounds out the `ISleepStrategy` concept by showing some possible other implementations.